hail/src/main/scala/is/hail/io/TextMatrixReader.scala

```diff
@@ -286,7 +286,7 @@ case class TextMatrixReader(
     val rvd = if (tr.dropRows)
       RVD.empty(sc, requestedType.canonicalRVDType)
     else
-      RVD.unkeyed(PStruct.canonical(requestedType.rowType).setRequired(true), rdd)
+      RVD.unkeyed(PStruct.canonical(requestedType.rowType).setRequired(true).asInstanceOf[PStruct], rdd)
```

Comment:
I definitely don't like all of these `asInstanceOf`. We'll probably never actually clean them up, and I'm not gating approval on it, but we should try to make them unnecessary through changes in `setRequired`.